src/sentry/discover/compare_timeseries.py

```diff
@@ -0,0 +1,172 @@
+
+    query_parts = QueryParts(selected_columns=[aggregate], query=query, equations=[], orderby=[])
+    query_parts = translate_mep_to_eap(query_parts)
+
+    results = run_timeseries_query(
+        snuba_params,
+        query_string=query_parts["query"],
+        y_axes=query_parts["selected_columns"],
+        referrer="job-runner.compare-timeseries",
+        granularity_secs=time_window,
+        config=SearchResolverConfig(),
+    )
+
+    return TSResultForComparison(result=results, agg_alias=query_parts["selected_columns"][0])
+
+
+def make_snql_request(
+    query: str,
+    aggregate: str,
+    time_window: int,
+    on_demand_metrics_enabled: bool,
+    snuba_params: SnubaParams,
+) -> TSResultForComparison:
+    query = apply_dataset_query_conditions(SnubaQuery.Type.PERFORMANCE, query, None)
+
+    results = timeseries_query(
+        [aggregate],
+        query,
+        snuba_params=snuba_params,
+        rollup=time_window,
+        referrer="job-runner.compare-timeseries",
+        on_demand_metrics_enabled=on_demand_metrics_enabled,
+        on_demand_metrics_type=MetricSpecType.SIMPLE_QUERY,
+        zerofill_results=True,
+    )
+
+    return TSResultForComparison(result=results, agg_alias=get_function_alias(aggregate))
+
+
+def align_timeseries(snql_result: TSResultForComparison, rpc_result: TSResultForComparison):
+    aligned_results: dict[str, Any] = defaultdict(lambda: {"rpc_value": None, "snql_value": None})
+
+    def fill_aligned_series(data: SnubaTSResult, alias: str, key: str):
+        for element in data.data:
+            element_value = element.get(alias) or 0
+            element_time = element["time"]
+            aligned_results[element_time][key] = float(element_value)
+
+    fill_aligned_series(snql_result["result"].data, snql_result["agg_alias"], "snql_value")
+    fill_aligned_series(rpc_result["result"].data, rpc_result["agg_alias"], "rpc_value")
+
+    return aligned_results
+
+
+def assert_timeseries_close(aligned_timeseries):
+    mismatches: dict[int, dict[str, float]] = {}
+    missing_buckets = 0
+    for timestamp, values in aligned_timeseries.items():
+        rpc_value = values["rpc_value"]
+        snql_value = values["snql_value"]
+        if rpc_value is None or snql_value is None:
+            missing_buckets += 1
+            continue
+
+        # If the sum is 0, we assume that the numbers must be 0, since we have all positive integers. We still do
+        # check the sum in order to protect the division by zero in case for some reason we have -x + x inside of
+        # the timeseries.
+        if rpc_value + snql_value == 0:
+            continue
+
+        mismatch = abs(rpc_value - snql_value)
+        average = (rpc_value + snql_value) / 2
+        diff = mismatch / average
+
+        if diff > 0.05:
+            mismatches[timestamp] = {
+                "rpc_value": rpc_value,
+                "snql_value": snql_value,
+                "mismatch_percentage": mismatch,
```

Comment:
This won't be the percentage like the key suggests, did you want to put `diff` here? I think the percentage will give us a good idea of how "off" we are using a normalized measure we can compare across the board

Reply:
oh good catch, i wanted to put diff here